File: Python/bee 1040.py
n1, n2, n3, n4 = map(float, input("").split(" "))
media = (n1 * 2 + n2 * 3 + n3 * 4 + n4)/10

print(f"Media: {media:.1f}")

if media >= 7:
    print("Aluno aprovado.")
elif media < 5:
    print("Aluno reprovado.")
else:
    print("Aluno em exame.")
    n5 = float(input(""))
    print(f"Nota do exame: {n5}")
    mediaFinal = (n5 + media)/2
    if mediaFinal >= 5:
        print("Aluno aprovado.")
    else:
        print("Aluno reprovado.")
    print(f"Media final: {mediaFinal:.1f}")

File: Ruby/bee 1040.rb
n1, n2, n3, n4 = gets.split(" ").map { |n| n.to_f }

media = (n1 * 2 + n2 * 3 + n3 * 4 + n4) / 10

puts sprintf("Media: %.1f", media)

if media >= 7
  puts "Aluno aprovado."
elsif media < 5
  puts "Aluno reprovado."
else  
  puts "Aluno em exame."
  n5 = gets.to_f
  puts sprintf("Nota do exame: %.1f", n5)
  media_final = (media+n5) / 2
  puts media_final > 5 ? "Aluno aprovado." : "Aluno reprovado."
  puts sprintf("Media final: %.1f", media_final)
end